cupy/_core/_carray.pxd

```diff
@@ -7,7 +7,11 @@ from cupy.cuda cimport function
 ctypedef vector.vector[Py_ssize_t] shape_t
 ctypedef vector.vector[Py_ssize_t] strides_t
 
-DEF MAX_NDIM = 25
+# this matches NPY_MAXDIMS
+DEF MAX_NDIM = 32
+cdef inline int get_max_ndim():
+    # we can't access module-level constants due to Cython's limitation...
+    return MAX_NDIM
```

Comment:
Not tested, but I guess this should work?
```
cdef enum: MAX_NDIM = 32
```

Reply:
This works indeed! Thanks. I was scratching my head when hitting https://github.com/cython/cython/issues/4369, knowing I knew a way that it works but couldn't recall...